File: content/cpp/templates/fibonacci.cpp
// Fibonacci Sequence

#include <iostream>
#include <vector>
using namespace std;

// Iterative version
vector<int> fibonacci(int n) {
    vector<int> fib;
    if (n <= 0) return fib;
    
    fib.push_back(0);
    if (n == 1) return fib;
    
    fib.push_back(1);
    for (int i = 2; i < n; i++) {
        fib.push_back(fib[i-1] + fib[i-2]);
    }
    return fib;
}

// Recursive version
int fibRecursive(int n) {
    if (n <= 1) return n;
    return fibRecursive(n-1) + fibRecursive(n-2);
}

int main() {
    int n = 10;
    
    cout << "First " << n << " Fibonacci: ";
    vector<int> fib = fibonacci(n);
    for (int num : fib) {
        cout << num << " ";
    }
    cout << endl;
    
    cout << "Fib(" << n << "): " << fibRecursive(n) << endl;
    
    return 0;
}

File: content/python/templates/fibonacci.py
# Fibonacci Sequence

def fibonacci(n):
    """Generate first n Fibonacci numbers."""
    if n <= 0:
        return []
    if n == 1:
        return [0]
    
    fib = [0, 1]
    for i in range(2, n):
        fib.append(fib[i-1] + fib[i-2])
    return fib

# Recursive version
def fib_recursive(n):
    """Get nth Fibonacci number recursively."""
    if n <= 1:
        return n
    return fib_recursive(n-1) + fib_recursive(n-2)

# Test
n = 10
print(f"First {n} Fibonacci: {fibonacci(n)}")
print(f"Fib({n}): {fib_recursive(n)}")
